### PR TITLE
docs: custom directive examples and docs page

### LINKED ISSUE #874: How to implement custom graphql directives with mercurius, is there a guide or minimal example?
How to implement custom graphql directives with mercurius, is there a guide or minimal example?

### PR DESCRIPTION
**Pull request by brainrepo**
This pr adds a doc page and an example of implementing a custom directive.
After a few iterations, to keep this example compact and avoid working with the AST, I decided to use two libraries, `@graphql-tools/schema` and `@graphql-tools/utils`.

I also added some comments in the example file, but I am unsure whether to keep them.

closes #874

### CODE REVIEW

**Review by simoneb — COMMENTED** [review]
let's call this redaction rather than censorship :) See e.g. bottom of this page https://www.fastify.io/docs/latest/Reference/Logging/

File: docs/custom-directive.md
Comment (simoneb) [inline docs/custom-directive.md:10]: I'd link to some docs explaining what possible locations there are besides FIELD_DEFINITION

File: docs/custom-directive.md
Comment (simoneb) [inline docs/custom-directive.md:18]: I'd clarify that this is the actual implementation of what the directive does

File: docs/custom-directive.md
Comment (simoneb) [inline docs/custom-directive.md:32]: what is the find property from the directive?

File: docs/custom-directive.md
Comment (simoneb) [inline docs/custom-directive.md:40]: for the sake of a simple example, I'd not mention the ability to configure the behavior of the directive via context. It is still useful to know that you can do that with the context though, so maybe we should first show the basic example, then mention that you can do more advanced things by accessing the context and making the behavior of the directive configurable

File: docs/custom-directive.md
Comment (simoneb) [inline docs/custom-directive.md:63]: ah ok I see now what "find" is. I think we should keep this simpler. how about we apply the directive on the field we want to redact, rather than on a parent and specify the text to find and replace with? the example you've come up with is not realistic, in most cases you don't know what text you want to redact, but you know the field that may contain that text and you want to redact the whole field. so in summary: apply the directive on a field and redact the whole field

**Review by simoneb — CHANGES_REQUESTED** [review]

**Comment by brainrepo** [comment]
Thank you, All your comments make sense, I am going to implement your suggestions.

File: docs/custom-directive.md
Comment (simoneb) [inline docs/custom-directive.md:15]: why bold this sentence?

File: docs/custom-directive.md
Comment (simoneb) [inline docs/custom-directive.md:40]: oh no don't list them all here, link to the docs!

File: examples/custom-directive.js
Comment (simoneb) [inline examples/custom-directive.js:63]: let's call this redact

**Review by simoneb — COMMENTED** [review]
Looks better but I still think that the use case you're using is not fitting. Maybe It wasn't clear in my previous comment but you can't redact something that you don't know what will be. You will need to redact the whole field instead. So I imagine the @redact directive to be applied to all fields of a type which possibly contain sensitive information, and they will redact the content of the whole field. Does this make sense now?

File: docs/custom-directive.md
Comment (simoneb) [inline docs/custom-directive.md:38]: display the link using markdown syntax, not a raw inline link please

File: docs/custom-directive.md
Comment (simoneb) [inline docs/custom-directive.md:57]: these regexps don't need to live inside this function

File: docs/custom-directive.md
Comment (simoneb) [inline docs/custom-directive.md:124]: what's all these new lines?

File: examples/custom-directive.js
Comment (simoneb) [inline examples/custom-directive.js:44]: add some blank lines here and there for readability please

**Review by simoneb — APPROVED** [review]
some comments, but looking better now, thanks

**Comment by brainrepo** [comment]
I implemented the suggestions

**Review by simoneb — APPROVED** [review]

**Comment by simoneb** [comment]
@mcollina this looks good to me but I'd like another pair of eyes before we merge this

**Review by mcollina — CHANGES_REQUESTED** [review]
Please add it to https://github.com/mercurius-js/mercurius/blob/master/docsify/sidebar.md

File: docs/custom-directive.md
Comment (mcollina) [inline docs/custom-directive.md:50]: We should document somewhere what version of graphql-tools this guide refers to `graphql-tools`.

I would also prefer to not use graphql-tools here or show both examples.

Comment (simoneb) [reply docs/custom-directive.md]: I agree with Mauro, we have discussed in detail and replicating the features coming from the graphql-tools library for the sake of this example is just not worth it, and quite frankly in general it's not worth it. Nevertheless since this issue comes up quite often, we'll come up with a couple of ideas that we'll open as issues in this repo:

- one specific to this request, a way built-into mercurius (or a plugin) to make it easier to define and attach custom directives
- one generic about creating a new library alternative to graphql-tools with similar features

Comment (brainrepo) [reply docs/custom-directive.md]: Yes, I have provided a more detailed argument on this issue.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**
